### DISCUSSION TITLE
Different parquet encodings

### DISCUSSION THREAD

**Discussion by jtbricker** [slingdata-io/sling-cli#286] [Q&A] [answered]
Is it currently possible to configure sling to write to parquet in different encodings.

It seems like the default encoding used is DELTA_LENGTH_BYTE_ARRAY, which is not supported when reading into Redshift Spectrum.

**Comment by flarco** [comment]
Hi, no it's not currently possible.

**Reply by jtbricker** [reply]
Its interesting that this encoding issue is happening since I imagine that the same data would be produced in the intermediate AWS Bucket if I were to do a direct transfer from BigQuery to Redshift. I might have to test that approach.   I'm still learning the tools so I appreciate you taking the time to answer

**Reply by flarco** [reply, marked as answer]
Yea, CSV is still the most compatible format. Parquet can have compatibility issues depending on the engine reading/writing it. BigQuery, for example, cannot export a JSON column into a Parquet file. You'd just have to manually cast it to a string. But it exports a JSON column to CSV no problem. So that's why Sling uses CSV as the temporary file format for bulk loading into Redshift/BigQuery (via COPY).

**Reply by jtbricker** [reply]
That makes a lot of sense.  Thanks for the extra context
